app/controllers/sites/companies/company/edit.js

```diff
@@ -0,0 +1,16 @@
+import Ember from 'ember';
+
+export default Ember.Controller.extend({
+  actions: {
+    updateCompany() {
+      const name = this.get('name');
+      const company = this.get('model');
+      company.set('name', name);
+      company.save();
```

Comment:
You should make this action consumable by a closure action and likely pass those values to the action. Take a look at an example of this in an app that I'm working on https://github.com/PairRequest/frontend/blob/master/app/components/login-form/component.js

The pattern is:
1. The context(ie... the controller) provides an action that knows what when object is saved
2. The action returns a promise
3. The action is passed into the form component as a closure action
4. In side of the form component, define an action that you trigger inside of component's layout
5. This action performs validation and if passed, calls the passed in closure action
6. Passed in closure action return the promise which is handled inside of the form

You should follow this pattern here.